File: BackEnd/Service/Models/InterviewModel.cs
namespace Service.Models
{
    public class InterviewModel
    {
        public Guid InterviewId { get; set; }

        public Guid InterviewerId { get; set; }

        public Guid RecruiterId { get; set; }

        public Guid ApplicationId { get; set; }

        public int? Company_Status { get; set; } 

        public int? Candidate_Status { get; set; } 

        public string? Notes { get; set; } 

        public int? Priority { get; set; } 

        public bool? IsDeleted { get; set; } = false;

        public Guid? ResultId { get; set; }
        public string? AddressOrStartURL { get; set; }
        public string? DetailLocationOrJoinURL { get; set; }
        public DateTime? MeetingDate { get; set; }
        public TimeSpan? StartTime { get; set; }
        public TimeSpan? EndTime { get; set; }

        public ApplicationModel Application { get; set; } 

        public InterviewerModel Interviewer { get; set; } 

        public RecruiterModel Recruiter { get; set; } = null!;

        public ResultModel Result { get; set; } 

        public ICollection<RoundModel> Rounds { get; set; } = new List<RoundModel>();
    }
}

File: FRONTEND/src/app/data/interview/interview.model.ts
import { Interview_CandidateStatus, Interview_CompanyStatus, Interview_Type } from "../../shared/enums/EInterview.model";
import { Application } from "../application/application.model"
import { Interviewer } from "../interviewer/interviewer.model"
import { Recruiter } from "../recruiter/recruiter.model"
import { Round } from "../round/round.module";

export class Interview {
	interviewId?: string;
	interviewerId?: string;
	endTime?: string;
	interviewer?: Interviewer;
	recruiterId?: string;
	recruiter?: Recruiter;
	applicationId?: string;
	application?: Application;
	company_Status?: Interview_CompanyStatus;
	candidate_Status?: Interview_CandidateStatus;
	notes?: string;
	priority?: string;
	isDeleted?: boolean;
	addressOrStartURL?: string;
	detailLocationOrJoinURL?: string;
	meetingDate?: Date;
	startTime?: string;
	interviewType?: Interview_Type;
	rounds?: Round[];
}

export class InterviewFilterModel {
	search?: string;
	sortString?: string;
	candidateStatus?: Interview_CandidateStatus;
	companyStatus?: Interview_CompanyStatus;
	fromTime?: string;
	toTime?: string;
	fromDate?: Date;
	toDate?: Date;
	interviewType?: Interview_Type;
	positionId?: string;
}
